From: Ben Hutchings <ben@decadent.org.uk>
Date: Wed, 20 Jun 2018 20:21:07 +0100
Subject: wireless: Avoid ABI change in 3.16.57
Forwarded: not-needed

Commit 657308f73e67 "regulatory: add NUL to request alpha2" enlarged
regulatory_request::alpha2 to make it a null-terminated string, so
that it could be used with netlink string functions.  Change it back
and convert to a null-terminated string in a local variable instead.

---
--- a/include/net/regulatory.h
+++ b/include/net/regulatory.h
@@ -78,7 +78,7 @@ struct regulatory_request {
 	int wiphy_idx;
 	enum nl80211_reg_initiator initiator;
 	enum nl80211_user_reg_hint_type user_reg_hint_type;
-	char alpha2[3];
+	char alpha2[2];
 	enum nl80211_dfs_regions dfs_region;
 	bool intersect;
 	bool processed;
--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@ -10393,10 +10393,14 @@ void nl80211_send_reg_change_event(struc
 			       NL80211_REGDOM_TYPE_INTERSECTION))
 			goto nla_put_failure;
 	} else {
+		u8 alpha2[3];
+
+		alpha2[0] = request->alpha2[0];
+		alpha2[1] = request->alpha2[1];
+		alpha2[2] = 0;
 		if (nla_put_u8(msg, NL80211_ATTR_REG_TYPE,
 			       NL80211_REGDOM_TYPE_COUNTRY) ||
-		    nla_put_string(msg, NL80211_ATTR_REG_ALPHA2,
-				   request->alpha2))
+		    nla_put_string(msg, NL80211_ATTR_REG_ALPHA2, alpha2))
 			goto nla_put_failure;
 	}
 
